protocol/vm/context.go

```diff
@@ -0,0 +1,31 @@
+package vm
+
+// Context contains the execution context for the virtual machine.
+//
+// By convention, variables of this type have the name context, _not_
+// ctx (to avoid confusion with context.Context).
+type Context struct {
+	VMVersion uint64
+	Code      []byte
+	Arguments [][]byte
+
+	TxVersion *uint64
+
+	BlockHash            *[]byte
+	BlockTimeMS          *uint64
+	NextConsensusProgram *[]byte
+
+	TxSigHash        *[]byte
```

Comment:
A slice is capable of being nil on its own, so if you want, the slice fields here don't have to be pointers. (Same as for the func field).